File: Classi/Classi/Classes/ViewControllers/ViewController.m
//
//  ViewController.m
//  Classi
//
//  Created by Mattia Contin  on 08/02/2017.
//  Copyright © 2017 Mattia Contin. All rights reserved.
//

#import "ViewController.h"

#import "Package.h"
#import "Courier.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    Package *packs[100];
    
    for(int i = 0; i < 100; i++) {
        packs[i] = [[Package alloc]initRandom];
        NSLog(@"%@", packs[i]);
    }
    
    int packsVolume;
    for(int i = 0; i < 100; i++) {
        packsVolume += [packs[i] volume];
    }
    
    Courier *courier = [[Courier alloc]initWithVolume:100];
    int neededCouriers = 1;
    while(packsVolume > [courier volume]) {
        packsVolume -= [courier volume];
        neededCouriers++;
    }
    if(![courier fits: packs]) {
        NSLog(@"troppi pacchi, servono %i corrieri", neededCouriers);
    }
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: SwiftUrlConnection/SwiftUrlConnection/ViewController.swift
//
//  ViewController.swift
//  SwiftUrlConnection
//
//  Created by Mattia Contin  on 28/02/2017.
//  Copyright © 2017 Mattia Contin. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        let url = URL(string: "https://jsonplaceholder.typicode.com/albums")
        
        let spinner = UIActivityIndicatorView(activityIndicatorStyle: UIActivityIndicatorViewStyle.gray)
        self.view.addSubview(spinner)
        spinner.startAnimating()
        
        spinner.hidesWhenStopped = true;
        spinner.center = view.center;
        
        
        let task = URLSession.shared.dataTask(with: url!) {
            (data, response, error) in
            
            guard error == nil else {
                print(error!)
                return
            }
            
            guard let data = data else {
                print("Data is empty")
                return
            }
            
            spinner.stopAnimating()
            let json = try! JSONSerialization.jsonObject(with: data, options: [])
            
            print(json)
        }
        
        task.resume()
 
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

